### PR TITLE
Completed array sorting method

### PR DESCRIPTION
**Pull request by Zaffierce**

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/insertionSort/InsertionSort.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package code401challenges.insertionSort;
+
+import java.util.Arrays;
+
+public class InsertionSort {
+
+    public static int[] insertionSort(int[] arr) {
+        if (arr.length == 0) {
+            throw new ArrayIndexOutOfBoundsException("Cannot sort an empty array");
```
Comment (jeffborda) [inline code401challenges/src/main/java/code401challenges/insertionSort/InsertionSort.java]: I might go with `IllegalArgumentException`, which Java docs say is "Thrown to indicate that a method has been passed an illegal or inappropriate argument." Mostly because the code you have would actually work with an empty array and not go out of bounds.

File: code401challenges/src/test/java/code401challenges/insertionSort/InsertionSortTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                sortMyArr.insertionSort(testArr));
+    }
+
+    //How to test for throwing errors.
+    //https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void insertionSortEmpty() {
+        int[] testArr = new int[]{};
+        assertEquals("This should throw an ArrayIndexOutOfBoundsException",
+                sortMyArr.insertionSort(testArr));
```
Comment (jeffborda) [inline code401challenges/src/test/java/code401challenges/insertionSort/InsertionSortTest.java]: I love that you are testing for the exception, and you can actually just delete the `assertEquals()` statement, and just call the method like you have below where it's commented out.

File: code401challenges/src/test/java/code401challenges/insertionSort/InsertionSortTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Test
+    public void insertionSortOne() {
+        int[] answer = new int[]{1};
+        int[] testArr = new int[]{1};
+        assertArrayEquals("This should only return one value",
+                answer,
+                sortMyArr.insertionSort(testArr));
```
Comment (jeffborda) [inline code401challenges/src/test/java/code401challenges/insertionSort/InsertionSortTest.java]: Great edge case to test for.
